File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/operation/delete/DeleteBody.java
package ru.citeck.ecos.records3.record.operation.delete;

import lombok.Data;
import ru.citeck.ecos.records3.RecordRef;

import java.util.ArrayList;
import java.util.List;

@Data
public class DeleteBody {

    private boolean debug = false;
    private List<RecordRef> records = new ArrayList<>();

    public void setRecords(List<RecordRef> records) {
        if (records != null) {
            this.records = new ArrayList<>(records);
        } else {
            this.records.clear();
        }
    }

    void setRecord(RecordRef record) {
        this.records.add(record);
    }
}

File: ecos-records/src/main/java/ru/citeck/ecos/records3/rest/v1/delete/DeleteBody.kt
package ru.citeck.ecos.records3.rest.v1.delete

import ru.citeck.ecos.records3.rest.v1.RequestBody
import ru.citeck.ecos.webapp.api.entity.EntityRef
import java.util.*

class DeleteBody : RequestBody() {

    var records: MutableList<EntityRef> = ArrayList<EntityRef>()
        private set

    fun setRecords(records: List<EntityRef>?) {
        if (records != null) {
            this.records = ArrayList<EntityRef>(records)
        } else {
            this.records = ArrayList()
        }
    }

    fun setRecord(record: EntityRef) {
        records.add(record)
    }
}
